botfront/imports/ui/components/nlu/common/UserUtteranceViewer.jsx

```diff
@@ -236,56 +236,52 @@ function UserUtteranceViewer(props) {
 
     return (
         <div className='utterance-viewer' data-cy='utterance-text'>
-            {textContent.map((element) => {
-                if (element.type === 'text') {
-                    return (
+            {textContent.map(element => (
+                <React.Fragment key={`${element.start}-${element.index}`}>
+                    {element.type === 'text' && (
```

Comment:
Why do we need a Fragment here ?
minor: imo using if is more readable that '&&' , but that's my opinion so make the change if you feel like it

Reply:
The fragment is to hold the React key in one place, instead of writing it three times. And that's also why I had to change the ifs.